### PR TITLE
Tx iter 4083 v7

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/7089

Describe changes:
- some optimizations for tx iterations

#11285 with persistent storage in the flow AppLayerParserState

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21553

File: src/app-layer-parser.c [hunk 1]
```diff
@@ -144,6 +144,10 @@ struct AppLayerParserState_ {
     AppLayerDecoderEvents *decoder_events;
 
     FramesContainer *frames;
+
+    uint64_t *tx_id_free_list;
```
Comment (victorjulien) [inline src/app-layer-parser.c:148]: this is per flow, which will use way too much memory. It should be per thread.

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/app-layer-parser.c (hunk 1 above)
Comment (catenacyber) [reply src/app-layer-parser.c:148]: Which structure should own it ?
- `FlowWorkerThreadData` ?
- `ThreadVars` ?
- something else ?

Comment (victorjulien) [reply src/app-layer-parser.c:148]: Perhaps a new `AppLayerSomething...` struct that is a member of `FlowWorkerThreadData`, but we only pass the new struct to the relevant call(s).

Comment (catenacyber) [reply src/app-layer-parser.c:148]: Ok, I have something in next PR

**Pull request closed without merging**